src/MorphoAaveV3Adapter.ts

```diff
@@ -565,18 +570,32 @@ export class MorphoAaveV3Adapter extends MorphoAaveV3DataEmitter {
       })
     );
 
-    const [ethBalanceOrVoid, morphoRewardsOrVoid] = (await Promise.all(
-      promises
-    )) as [BigNumber, UserData["morphoRewards"] | null];
+    const [
+      ethBalanceOrVoid,
+      morphoRewardsOrVoid,
+      managerApprovalOrVoid,
+      stEthBalanceOrVoid,
+    ] = (await Promise.all(promises)) as [
+      BigNumber,
+      UserData["morphoRewards"] | null,
+      boolean | null,
+      StEthData | null
+    ];
```

Comment:
Couldn't we generate properly the promises instead of pushing to the array and retype-checking by hand? We could simply use a flat on an array?

Reply:
this is used to leverage on the multicall as much as possible. can you provide an example of your recommendation plz?